### DISCUSSION TITLE
Numpy style showing up as google style

### DISCUSSION THREAD

**Discussion by quantfreedom** [mkdocstrings/mkdocstrings#555] [Q&A]
I have to admit this has been the most frustrating process in the history of ever. There seems to be no examples anywhere on how to actually get the numpy style to work or any examples of what the default output would look like if you use numpy or google. I would expect something like this https://numpydoc.readthedocs.io/en/latest/example.html#example to be on the mkdocstrings site and an example of what your mkdocstrings section should look like in your yml file ... but there isn't ... i would also expect there to be examples of what your strings would look like if you choose table or list but again nothing.

I am just guessing because there is no actual documentation examples on how to do this properly but I am assuming this is the correct way to setup my doc strings to work with numpy 
![image](https://user-images.githubusercontent.com/95321908/231214170-0887870d-ce07-480f-8338-d1d6970cdd31.png)

i also tried it like this with no luck
![image](https://user-images.githubusercontent.com/95321908/231220848-61335249-9ef7-4491-9332-fc540fa4f9ca.png)

even though this is the usage section there seems to be no usage examples of numpy https://mkdocstrings.github.io/python/usage/#supported-docstrings-styles nor on the legacy you get nothing and reading the numpydocs's doc doesn't give use an understanding of how to actually use it in mkdocstrings

so i created a new file just to test the numpy example that you have at https://numpydoc.readthedocs.io/en/latest/example.html#example and the result i get is it looks like google 
![image](https://user-images.githubusercontent.com/95321908/231214837-25cdfbc0-a769-4c63-8b51-fa9e6ec4e396.png)

i have been searching the mkdocstrings docs for examples of anything concreate on how to actually change styles or sections styles or anything but there seems to be nothing. 

```
Package                    Version
-------------------------- -----------
aiodns                     3.0.0
aiohttp                    3.8.4
aiosignal                  1.3.1
ansi2html                  1.8.0
anyio                      3.6.2
appdirs                    1.4.4
argon2-cffi                21.3.0
argon2-cffi-bindings       21.2.0
arrow                      1.2.3
asttokens                  2.2.1
async-timeout              4.0.2
attrs                      22.2.0
autopep8                   2.0.2
backcall                   0.2.0
beautifulsoup4             4.11.2
black                      23.1.0
bleach                     6.0.0
blosc2                     2.0.0
ccxt                       2.9.13
certifi                    2022.12.7
cffi                       1.15.1
charset-normalizer         3.1.0
click                      8.1.3
colorama                   0.4.6
comm                       0.1.2
contourpy                  1.0.7
cryptography               39.0.2
csscompressor              0.9.5
cycler                     0.11.0
Cython                     0.29.34
dash                       2.9.1
dash-bootstrap-components  1.4.1
dash-bootstrap-templates   1.0.8
dash-core-components       2.0.0
dash-html-components       2.0.0
dash-table                 5.0.0
dateparser                 1.1.7
debugpy                    1.6.6
decorator                  5.1.1
defusedxml                 0.7.1
dill                       0.3.6
docstring-parser           0.15
executing                  1.2.0
fastjsonschema             2.16.3
Flask                      2.2.3
fonttools                  4.39.2
fqdn                       1.5.1
frozendict                 2.3.5
frozenlist                 1.3.3
ghp-import                 2.1.0
griffe                     0.25.5
h5py                       3.8.0
html5lib                   1.1
htmlmin                    0.1.12
idna                       3.4
imageio                    2.26.1
ipykernel                  6.21.3
ipython                    8.11.0
ipython-genutils           0.2.0
ipywidgets                 7.7.2
isoduration                20.11.0
itsdangerous               2.1.2
jedi                       0.18.2
Jinja2                     3.1.2
joblib                     1.2.0
jsmin                      3.0.1
jsonpointer                2.3
jsonschema                 4.17.3
jupyter_client             8.0.3
jupyter_core               5.2.0
jupyter-dash               0.4.2
jupyter-events             0.6.3
jupyter_server             2.4.0
jupyter_server_terminals   0.4.4
jupyterlab-pygments        0.2.2
jupyterlab-widgets         1.1.1
kaleido                    0.1.0.post1
kiwisolver                 1.4.4
llvmlite                   0.39.1
lxml                       4.9.2
Markdown                   3.3.7
MarkupSafe                 2.1.2
matplotlib                 3.7.1
matplotlib-inline          0.1.6
mergedeep                  1.3.4
mistune                    2.0.5
mkdocs                     1.4.2
mkdocs-autorefs            0.4.1
mkdocs-gen-files           0.4.0
mkdocs-literate-nav        0.6.0
mkdocs-material            9.1.2
mkdocs-material-extensions 1.1.1
mkdocs-minify-plugin       0.6.2
mkdocs-open-in-new-tab     1.0.2
mkdocs-section-index       0.3.5
mkdocs-video               1.4.0
mkdocstrings               0.21.2
mkdocstrings-python        0.8.3
msgpack                    1.0.5
multidict                  6.0.4
multitasking               0.0.11
mypy-extensions            1.0.0
nbclassic                  0.5.3
nbclient                   0.7.2
nbconvert                  7.2.9
nbformat                   5.7.3
nest-asyncio               1.5.6
notebook                   6.5.3
notebook_shim              0.2.2
numba                      0.56.4
numexpr                    2.8.4
numpy                      1.23.5
packaging                  23.0
pandas                     1.5.3
pandocfilters              1.5.0
parso                      0.8.3
pathspec                   0.11.0
pickleshare                0.7.5
Pillow                     9.4.0
pip                        23.0.1
platformdirs               3.1.1
plotly                     5.13.1
polars                     0.16.12
prometheus-client          0.16.0
prompt-toolkit             3.0.38
psutil                     5.9.4
pure-eval                  0.2.2
py-cpuinfo                 9.0.0
pyarrow                    11.0.0
pybit                      2.4.1
pycares                    4.3.0
pycodestyle                2.10.0
pycparser                  2.21
Pygments                   2.14.0
pymdown-extensions         9.10
pyparsing                  3.0.9
pyrsistent                 0.19.3
python-dateutil            2.8.2
python-json-logger         2.0.7
pytkdocs                   0.16.1
pytz                       2022.7.1
pytz-deprecation-shim      0.1.0.post0
pywin32                    305
pywinpty                   2.0.10
PyYAML                     6.0
pyyaml_env_tag             0.1
pyzmq                      25.0.0
regex                      2022.10.31
requests                   2.28.2
retrying                   1.3.4
rfc3339-validator          0.1.4
rfc3986-validator          0.1.1
schedule                   1.1.0
scikit-learn               1.2.2
scipy                      1.10.1
Send2Trash                 1.8.0
setuptools                 65.5.0
six                        1.16.0
sniffio                    1.3.0
soupsieve                  2.4
stack-data                 0.6.2
TA-Lib                     0.4.24
tables                     3.8.0
tabulate                   0.9.0
tenacity                   8.2.2
terminado                  0.17.1
threadpoolctl              3.1.0
tinycss2                   1.2.1
tomli                      2.0.1
tornado                    6.2
tqdm                       4.65.0
traitlets                  5.9.0
typing_extensions          4.5.0
tzdata                     2022.7
tzlocal                    4.3
uri-template               1.2.0
urllib3                    1.26.15
vectorbt                   0.24.5
watchdog                   2.3.1
wcwidth                    0.2.6
webcolors                  1.12
webencodings               0.5.1
websocket-client           1.5.1
websockets                 10.4
Werkzeug                   2.2.3
wheel                      0.40.0
widgetsnbextension         3.6.2
yarl                       1.8.2
yfinance                   0.2.13
```

I really wish a library about making documentation would actually have good documentation ... you would think that would be the case but seems to not be the case ... if anyone could help me that would be nice.

**Comment by pawamoy** [comment]
Hey @QuantFreedom1022, I'm happy to see that you didn't quit. Thank you for taking the time to open an issue.

When you open issues/discussions, try to paste text and code blocks rather than screenshots of your code: text makes issues searchable, and maintainers can copy-paste it to suggest fixes.

Your second try (second screenshot) seems correct to me. Then the rendering looks fine as well. I think you're confusing "docstring style" here with "how it looks". We commonly refer to "docstring styles" to distinguish the different standards available to format information in docstrings. So, the Google style is about writing docstrings like this:

```
Summary.

Arguments:
    arg1: Description.

Returns:
    Description.
```

...while the Numpydoc style is about writing them like this:

```
Summary.

Parameters
----------
arg1
    Description.

Returns
-------
Description.
```

These styles have nothing to do with how the docstring is eventually rendered by a variety of tools.

You can still customize the look and feel of your HTML pages using extra CSS, see https://www.mkdocs.org/user-guide/configuration/#extra_css or https://squidfunk.github.io/mkdocs-material/customization/?h=css#additional-css.

---

Now about your other comments. I understand that you are frustrated, but please keep in mind that until very recently, I've worked on mkdocstrings entirely on my free time, during evenings, week-ends and holidays. I've worked on it for more than 3 years now, giving away my work for free, to you and everyone else. Your feedback is very much welcome, but I'd appreciate if you could give it in a positive, [kind](https://snarky.ca/setting-expectations-for-open-source-participation/), constructive way.

Writing good documentation is hard. I did and will continue to do my best to improve mkdocstrings' docs. You can also contribute and send pull requests to improve them, if you want to.

**Comment by pawamoy** [comment]
I'll convert this issue as a discussion since I don't think there's an issue here. Don't hesitate to open other issues if you find some :slightly_smiling_face:

**Comment by quantfreedom** [comment]
Please add everything you said to your documentation .... something like ... if you choose google docstring styles you must write your docstrings like this (example) if you choose numpy you must write your docstrings like this (example) ... but no matter which docstyle you choose to write they will both end up looking like this on your website (example) ... if you want to change up how it looks you can do this and that with css code like if you want to move the description to start on the next line and want it to be indented you would do this (example)

your entire documentation needs more through examples ... and i too am working on a project and am making documentation as well and it is already more through and has more, examples including videos walking people through how to use it, with tons of other examples as well ... so how do i already have better documentation than a person making a site about documentation ?

but the fact that you are making a documentation website about making documentation without good documentation is very weird  ... the usability of it needs so much work ... but seeing as how you don't ever want to jump on calls with users and actually watch their experience of using your product shows that you don't really actually care about making good documentation so there is no point in giving any feedback on that ... 

Like think about it logically for a second ... how many installs have you got and what percent of those people are giving you feedback on your documentation ... then out of that feedback how many of those people are actually contributing to the code to do a pull request to help you fix it ... i would say probably .0003% of the time that happens ... then on top of that ... its probably for a very specific section ... so at this rate you will never ever ever ever ever know what the true user experience is like if you don't get over yourself and start walking through the docs with random people ... its just backed up by math 

what i suggest you do is hire someone that is a UX UI person so they can guide you if your not going to take feedback from users in real time. If i were to be giving you feedback on what i am going through using your website i would be here for years explaining it ... so you can't use working on this in my free time as an excuse because we all do that and there are some people who have great documentation and some that have trash ... 

but the common theme amongst the people who have trash documentation is their unwillingness to actually get the real user experience ... because 95% or more of people aren't going to leave feedback about your website ... they will just bounce of the site if it sucks so you will never know why they left if you don't actually walk through it with them. and also if someone already knows how to code and figures all this stuff out then you won't hear from them either ... 

so you are screwing yourself over by not actively trying to work with people because you will never know unless the random .005% of people actually want to give you feedback on one particular part of the documentation ... where what you really need is the start to finish experience of anyone ... if you don't have that then it will be impossible for you to know what is the most important parts to improve upon 

you need to get over yourself and do the things necessary to take your business to the next level if you want this to work or you are going to forever be stuck or grow slow as hell ... i have no idea how you don't get this but i guess you are just the typical person like that last guy i knew who had people constantly asking questions because the docs were so bad and refused to also get on a call to talk to people about their experience and is growing slow as all hell ... been working on his project for multiple multiple multiple years and growth is still 2 new sponsors a day even though he has been doing this for over 5 years ... i just don't get how people like you don't get it ... its such a simple thing to do ... but i guess that is why there are winners and losers in this world ... people don't know how to grow even if they have a good idea

**Comment by pawamoy** [comment]
I was going to answer point by point, but decided I won't. Your whole comment is very rude and dismissive of my work, and I really don't enjoy interacting with you. As an open-source developer providing my work for free, I owe you, or anyone else, nothing. You are behaving like what we call an *entitled user*. Please educate yourself about this. If you don't like my project, don't use it. Or fork it. Or write your own. I don't have time for this. I will use your comments as an example how **not** to behave as a user. You're free to open new issues/discussions if you're able to stop being rude, dismissive and constantly telling people what they should do or how. Locking this discussion.
